packages/tf-inventory/src/inventory.js

```diff
@@ -0,0 +1,95 @@
+'use strict';
+
+const aws = require('@cumulus/common/aws');
+const mergeWith = require('lodash.mergewith');
+const difference = require('lodash.difference');
+const stateFile = require('./stateFile');
+
+/**
+ * Merge 2 resource objects by key
+ * So { ecs: [1, 2, 3] } and { ecs: [4, 5] }
+ * would become { ecs: [1, 2, 3, 4, 5] }
+ * This assumes x and y are objects of lists
+ *
+ * @param {Object} x - resource object
+ * @param {Object} y - resource object
+ * @returns {Objects} - resource object of x and y combined
+ */
+function mergeResourceLists(x, y) {
+  return mergeWith(x, y, (xVal, yVal) => {
+    if (!xVal) {
+      return yVal;
+    }
+
+    if (!yVal) {
+      return xVal;
+    }
+
+    return xVal.concat(yVal);
+  });
+}
+
+/**
+ * For 2 resource objects of lists, return an object where each
+ * key has items that are only in the first list
+ * For example:
+ * { ecs: [1, 2, 3] } and { ecs: [1, 2, 4] }
+ * would return { ecs: [3] }
+ *
+ * @param {Object} x - resource object
+ * @param {Object} y - resource object
+ * @returns {Object}
+ */
+function resourceDiff(x, y) {
+  const keys = Object.keys(x);
+  const val = {};
+
+  keys.forEach((k) => {
+    val[k] = difference(x[k], y[k]);
+  });
+
+  return val;
+}
+
+/**
+ * Go through each state file and get resources. Return Object containing
+ * resources from all state files.
+ *
+ * @param {Array<string>} stateFiles - List of state file paths in the form
+ * bucket/key
+ * @returns {Object} - Object of resources from all state files
+ */
+async function listTfResources(stateFiles) {
+  const resourcePromises = stateFiles.map((sf) => stateFile.listResourcesForFile(sf));
+
+  const resources = await Promise.all(resourcePromises);
+
+  return resources.reduce(mergeResourceLists);
+}
+
+async function listAwsResources() {
```

Comment:
No JSDoc

Reply:
This is not exported.

Reply:
I still think it's useful to have them on non-trivial functions that are not exported because it documents the expected input and output types.. _"Something something statically-typed language would self-document."_ 😅